deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -0,0 +1,614 @@
+""" Run Farm management. """
+
+from __future__ import annotations
+
+import re
+import logging
+import abc
+from fabric.api import prefix, local, run, env, cd, warn_only, put, settings, hide # type: ignore
+from fabric.contrib.project import rsync_project # type: ignore
+import time
+from os.path import join as pjoin
+
+from util.streamlogger import StreamLogger
+from awstools.awstools import terminate_instances, get_instance_ids_for_instances
+
+from typing import List, Dict, Optional, Union, TYPE_CHECKING
+if TYPE_CHECKING:
+    from mypy_boto3_ec2.service_resource import Instance as EC2InstanceResource
+    from runtools.firesim_topology_elements import FireSimSwitchNode, FireSimServerNode
+    from runtools.run_farm import Inst
+    from awstools.awstools import MockBoto3Instance
+
+rootLogger = logging.getLogger()
+
+class InstanceDeployManager(metaclass=abc.ABCMeta):
```

Comment:
In other words, this class does everything after being given a hostname to access an instance.